core/src/apps/webauthn/credential.py

```diff
@@ -243,6 +314,16 @@ def private_key(self) -> bytes:
             return b""
         return self.node.private_key()
 
+    def public_key(self) -> bytes:
+        return nist256p1.publickey(self.private_key(), False)
+
+    def sign(self, data: Iterable[bytes]) -> bytes:
```

Comment:
The method is nearly duplicated in `Fido2Credential.sign` and `basic_attestation_sign`. Maybe it's worth refactoring it somehow.

Reply:
Done.

Reply:
I was thinking of the following function, but OK:
```
def sign(key, data):
  dig = hashlib.sha256()
  for segment in data:
    dig.update(segment)
    sig = nist256p1.sign(key, dig.digest(), False)
    return der.encode_seq((sig[1:33], sig[33:]))
```